winrm_plugin/tasks.py

```diff
@@ -0,0 +1,107 @@
+# standard library imports
+import base64
+import os.path
+
+# installed libraries imports
+import winrm
+
+# our library imports
+from cloudify import ctx
+from cloudify.decorators import operation
+
+
+@operation
+def run_script(address, username, password, process, local_file_path,
+               delete_after_running=True, remote_script_path=None, port=5985,
+               protocol="http", **kwargs):
+
+    remote_script_file_name = "\\temp" + os.path.splitext(local_file_path)[1]
+
+    conn = get_conn(protocol, address, password, username, port)
+
+    remote_shell_id = get_remote_shell_id(conn)
+
+    powershell_path = define_script_path(remote_script_path, False)
+
+    encoded_script = create_script_creation_command(
+            local_file_path, powershell_path, remote_script_file_name)
+
+    # copy the script file
+    ctx.logger.info('Create script file on remote machine')
+    run_remote_command(remote_shell_id, 'powershell', '-encodedcommand',
+                       ' ' + encoded_script, conn)
+
+    cmd_path = define_script_path(remote_script_path)
+    ctx.logger.info('Runing the script on remote machine')
+    run_remote_command(remote_shell_id, process, cmd_path,
+                       remote_script_file_name, conn)
+
+    if delete_after_running:
+        ctx.logger.info('Removing script file from remote machine')
+        run_remote_command(remote_shell_id, 'del', cmd_path,
+                           remote_script_file_name, conn)
+
+
+@operation
+def run_commands(commands, address, username, password,
+                 process, port=5985, protocol="http", **kwargs):
+
+    conn = get_conn(protocol, address, password, username, port)
+
+    remote_shell_id = get_remote_shell_id(conn)
+
+    for command in commands:
+        encode_command = create_encoded_command(command)
+        ctx.logger.info('running command: {0}'.format(encode_command))
+        run_remote_command(remote_shell_id, process,  encode_command, " ", conn)
+
+
+def define_script_path(remote_script_path, is_cmd=True):
+
+    tmp_env_var = '%TEMP%' if is_cmd else '$env:TEMP'
```

Comment:
Consider using this:

https://docs.python.org/2/library/tempfile.html#tempfile.tempdir